.github/workflows/release.yml

```
@@ -0,0 +1,107 @@
+name: Release Workflow
+
+on:
+  push:
+    tags:
+      - 'v*'  # Trigger on tag push like v0.1.0
+  workflow_dispatch:
+    inputs:
+      type:
+        description: 'Type of release (major/minor/patch)'
+        required: true
+        default: 'minor'
+      dry_run:
+        description: 'Dry run (true/false)'
+        required: true
+        default: 'true'
+      skip_tests:
+        description: 'Skip tests (true/false)'
+        required: true
+        default: 'false'
+
+permissions:
+  contents: write
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build_and_test:
+    name: Build and Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Set up Rust
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          toolchain: stable
+
+      - name: Cache Cargo
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/index
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Build Project
+        run: cargo build --release --verbose
+
+      - name: Run Tests
+        if: ${{ github.event.inputs.skip_tests != 'true' }}
+        run: cargo test --verbose
+
+      - name: Save Release Binary
+        if: startsWith(github.ref, 'refs/tags/v')
+        run: |
+          mkdir -p release-artifacts
+          cp target/release/didcomm-mediator release-artifacts/
```

You may want to ensure the binary has the correct executable permissions before uploading
```suggestion
      - name: Save Release Binary
        if: startsWith(github.ref, 'refs/tags/v')
        run: |
          mkdir -p release-artifacts
          cp target/release/didcomm-mediator release-artifacts/
          chmod +x release-artifacts/didcomm-mediator
```

---

Okay on it

---

Done